activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/starter/tests/cmdendpoint/CommandEndpointIT.java

```diff
@@ -0,0 +1,421 @@
+import org.springframework.hateoas.PagedResources;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource("classpath:application-test.properties")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+@EnableBinding(MessageClientStream.class)
+public class CommandEndpointIT extends KeycloakEnabledBaseTestIT {
+
+    @ClassRule
+    public static RabbitTestSupport rabbitTestSupport = new RabbitTestSupport();
+
+    @Autowired
+    private MessageChannel myCmdProducer;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private Tracer tracer;
+
+    public static final ParameterizedTypeReference<PagedResources<Task>> PAGED_TASKS_RESPONSE_TYPE = new ParameterizedTypeReference<PagedResources<Task>>() {
+    };
+
+    private Map<String, String> processDefinitionIds = new HashMap<>();
+
+    public static final String PROCESS_DEFINITIONS_URL = "/v1/process-definitions/";
+    public static final String PROCESS_INSTANCES_RELATIVE_URL = "/v1/process-instances/";
+    private static final String TASKS_URL = "/v1/tasks/";
+
+    private static final String SIMPLE_PROCESS = "SimpleProcess";
+
+    private static AtomicBoolean startedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean suspendedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean activatedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean claimedTaskAck = new AtomicBoolean(false);
+    private static AtomicBoolean releasedTaskAck = new AtomicBoolean(false);
+    private static AtomicBoolean completedTaskAck = new AtomicBoolean(false);
+
+
+
+    private StartProcessInstanceCmd startProcessInstanceCmd;
+
+    private SuspendProcessInstanceCmd suspendProcessInstanceCmd;
+
+    private static String testProcessInstanceId;
+
+    @Before
+    public void setUp() throws Exception {
+        keycloaktestuser = "hruser";
+        //don't need to set password as same password as testuser
+        accessToken = authenticateUser();
+
+        // Get Available Process Definitions
+        ResponseEntity<PagedResources<ProcessDefinition>> processDefinitions = getProcessDefinitions();
+        assertThat(processDefinitions.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        assertThat(processDefinitions.getBody().getContent()).hasSize(6); //if a new definition is added then this is expected to be increased
```

Comment:
In my opinion tests should be written in a such way that having a new process definition in the classpath does not impact them. I really think that this is going to be hard to maintain if every time that you need to add a new process you need to modify a dozen of `assertThat(processDefinitions.getBody().getContent()).hasSize(6)`.